### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by Engil**
Heya,

This extensive word salad 🥗  is a first shot as a README for the project.

It is far from being perfect, and feel free to shotgun any changes to it directly on this branch, or in the main branch if you feel like merging it right away.

There's also a concerning lack of emojis in it, which we may address later on. 🦦 
This PR is best reviewed on a commit-per-commit basis. (there's only one)

Also, I introduced the following wiki page: https://github.com/pitag-ha/cardio-crumble/wiki/MIDI-Setup

I tested the MacOS part, could you add your workflow on Linux if you have time? 😊 

Thanks, and see ya!
🐬

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+Do note that if the program you are running cardio-crumble on needs to be passed parameters as well, you need to make sure you properly segment your command line call, so that `dune` knows to which program a parameter belongs:
+
+
+```shell
+dune exec -- bin/main.exe stat_engine --device_id=1 --bpm=60 -- my_program -- my_program_parameters
+```
+
+### Engines and optional parameters
+
+There exists current two engines in cardio-crumble:
```
Comment (pitag-ha) [inline README.md:87]:
```suggestion
There are currently two engines in cardio-crumble:
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+You can list there by running the `cardio-crumble` executable with an engine name, followed by the `--help` option.
+
+```shell
+dune exec bin/main.exe -- stat_engine --help
+```
+
+Notable options:
+
+- `--scale` Allows you to select a musical [scale](https://en.wikipedia.org/wiki/Scale_(music)) to which `cardio-crumble` will stick to when generating notes. Available options are `minor`, `blue`, `major`, and the experimental `nice` scale.
```
Comment (pitag-ha) [inline README.md:102]:
```suggestion
- `--scale` Allows you to select a musical [scale](https://en.wikipedia.org/wiki/Scale_(music)) to which `cardio-crumble` will stick to when generating notes. Available options are `minor`, `blue`, `major`, and the ~~...nice...~~ experimental `nice` scale.
```

Does `~~` work in README's?

**Review by pitag-ha — APPROVED** [review]
Nice, thanks a lot!

I agree on the emoji part :neckbeard: Let's leave it for later, though...

**Comment by Engil** [comment]
I applied your suggestions, thank you for the thorough review!
I will be merging.

**Pull request merged**